km/km_filesys.c

```diff
@@ -2273,7 +2273,7 @@ uint64_t km_fs_select(km_vcpu_t* vcpu,
 uint64_t km_fs_poll(km_vcpu_t* vcpu, struct pollfd* fds, nfds_t nfds, int timeout)
 {
    for (int i = 0; i < nfds; i++) {
-      if (km_fs_g2h_fd(fds[i].fd, NULL) < 0) {
+      if ((fds[i].fd != 0xffffffff) && (km_fs_g2h_fd(fds[i].fd, NULL) < 0)) {
```

Comment:
The poll man page says a negative fd (not just -1) means that entry in the pollfd array should be ignored and that revent should be set to zero.